crates/observe/src/request_id.rs

```diff
@@ -39,6 +40,21 @@ where
     REQUEST_ID.scope(id, scope).await
 }
 
+/// Spawns a new task. If the current task has a request_id it gets copied
+/// over into the spawned task to allow for tracing requests across task
+/// boundaries.
+pub fn spawn_task_with_request_id<F>(future: F) -> JoinHandle<F::Output>
+where
+    F: Future + Send + 'static,
+    F::Output: Send + 'static,
+{
+    if let Some(id) = get_task_local_storage() {
+        tokio::task::spawn(REQUEST_ID.scope(id, future))
+    } else {
+        tokio::task::spawn(future)
```

Comment:
If the request id is not found maybe we should at least log that, as the intention of the caller of spawn_task_with_request_id() function is not fulfilled.